### DISCUSSION TITLE
Should we add a second stream struct to include the recipient?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/flow#161] [Ideas]
Similar to lockup contracts, where we've added a struct that includes the recipient, should we add here one as well?

Context:
- Common struct (storage): https://github.com/sablier-labs/v2-core/blob/cf246ef5199289539426ee19476f3ef9ebf0556d/src/types/DataTypes.sol#L82-L97
- `StreamLL` used only in `getStream`: https://github.com/sablier-labs/v2-core/blob/cf246ef5199289539426ee19476f3ef9ebf0556d/src/types/DataTypes.sol#L254-L269

**Comment by andreivladbrg** [comment]
RFC @smol-ninja

**Comment by smol-ninja** [comment]
What will it be used for?

**Reply by andreivladbrg** [reply]
`getStream` getter - retrieve all stream data within one call

**Reply by smol-ninja** [reply]
That looks like a useful API to me. I am in favour of it.

**Comment by smol-ninja** [comment]
I changed my mind about it. I think this would be overkill. The same data can be fetched using `flow.getStream(streamId)` and `flow.getRecipient(streamId)` very easily. The benefit this second struct would offer is very less compared to its implementation and test (even though its low effort, the lines of code would not be so less).

**Comment by smol-ninja** [comment]
@andreivladbrg if you have no comment on this, could we please close this discussion?

**Comment by andreivladbrg** [comment]
Yeah, not worth it for now
